### PR TITLE
Document hooks in README

### PR DESCRIPTION
**Pull request by razor-x**
- docs: Document hooks in README
- Export SeamProvider in hooks entrypoint

### CODE REVIEW

**Review by dawnho — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -86,6 +86,57 @@ export const App = () => {
...
+
+Import hooks directly from `@seamapi/react` or `@seamapi/react/hooks`.
+For most applications, these imports are equivalent,
+however if you are only using the hooks, we recommend importing from `@seamapi/react/hooks`
+as it may enable performance or bundling improvements depending on your build system.
+
+Hooks must be used inside the `SeamProvider`.
+They are well-typed and follow a uniform API.
+
+```ts
```
Comment (xplato) [inline README.md:102]: Should this be `tsx` or is there no support for that?

**Review by xplato — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (razor-x) [reply README.md:102]: Oh nice! Let me try 👍🏻

Comment (razor-x) [reply README.md]: I was wondering why the highlighting was not perfect ❤️  [`1f2a944` (#260)](https://github.com/seamapi/react/pull/260/commits/1f2a9445863f955d132dc40628bcd49914e58418)

Comment (xplato) [reply README.md]: Awesome, good to know!

**Pull request merged**